packages/adapter-auto/index.js

```diff
@@ -1,9 +1,7 @@
 import { execSync } from 'node:child_process';
-import { pathToFileURL } from 'node:url';
-import { resolve } from 'import-meta-resolve';
 import { adapters } from './adapters.js';
-import { dirname, join } from 'node:path';
-import { existsSync } from 'node:fs';
+import path from 'node:path';
+import fs from 'node:fs';
 import process from 'node:process';
 
 /** @type {Record<string, (name: string, version: string) => string>} */
```

Comment:
This logic exists in `package-manager-detector` (used by svelte CLI, changesets, and publint). It might be good to use here too as it supports Deno, which I notice we're not supporting here

Reply:
maybe we could just follow this PR up with one that adds Deno support? Or is there a reason to strongly prefer adding the dependency?

Reply:
yeah, we probably could. I think the main reason to possibly prefer the dependency is that getting some of the commands right for Deno has been maddeningly difficult. even now there's an issue open that the install command is not working correctly: https://github.com/antfu-collective/package-manager-detector/issues/51